src/motion_planning/src/plan_aak.cpp

```diff
@@ -0,0 +1,744 @@
+   void get_data::get_origin(float x , float y)
+  {    
+        pairs position;
+
+        position = get_grid_pos(x,y);
+
+         start_row = position.first;
+         start_col  = position.second; 
+
+         if (map[start_row][start_col].cell_status == 100)
+         {
+
+             cout<<" start position is not reachable ---mission fail "<< endl ;
+             //exit(1);
+
+         }
+
+   }
+
+
+
+
+  // function to get data from csv file
+
+ void get_data::data()
+  
+  {
+
+    int count = 0;
+    ifstream myFile;
+    myFile.open("/home/satyam/turtlebot/out2.csv");
+
+    while (myFile.good() && (count< LENGTH) )
+    {
+      string y;
+      getline(myFile, y);
+      int num = stoi(y);
+      x[count] = num;
+      count ++;
+    }
+     
+    myFile.close();
+    reshape(x);
+
+  }
+
+
+
+//function to shape arrays
+
+void get_data::reshape(int x[LENGTH])
+{
+
+  cout<<" entered reshape successfully"<<endl;
+  int inc = 0;
+ 
+
+for (int i = ROWS-1; i>=0; i--)
+   {
+  
+    for (int j=0; j<COLUMNS; j++)
+    {
+      
+      if( x[inc] < 0 || x[inc]==100)
+      {
+        map[i][j].cell_status = 100;
+        map[i][j].cell_charc = '*';
+      }
+
+      if( x[inc] == 0)
+      {
+        map[i][j].cell_status = 0;
+        map[i][j].cell_charc = '0';
+      }
+      inc++;
+    
+     }
+   }
+
+
+  //  for (int i=0; i<ROWS; i++)
+  //  {
+  //    cout << endl;
+  //   for (int j=0; j<COLUMNS; j++)
+  //   {
+       
+  //     cout<< map[i][j].cell_charc;
+  //   }
+  // }
+   cout<< "reshaped successfully"<<endl;
+
+}
+
+float  get_data::heuristics(int current_node_row,int current_node_col,int goal_row,int goal_col)
+{
+  
+   float heur_distance = sqrt(pow((current_node_row-goal_row),2)+ pow((current_node_col-goal_col),2)) ;
+   // cout<< heur_distance<<endl;
+   return heur_distance ;
+}
+
+
+
+
+
+
+
+// function for astar main file
+void get_data::astar()
+
+{
+  cout << "entering astar"<< endl;
+  
+  for(int i=0;i<ROWS;i++)
+  {
+      //cout<<endl;
+      for(int j=0;j<COLUMNS;j++)
+      {
+        map[i][j].move_cost = 0;
+        map[i][j].heur_cost = max;
+        map[i][j].cost = max ;
+        map[i][j].visited = false ;
+        map[i][j].parent_node_row = -1;
+        map[i][j].parent_node_col = -1;
+
+      }
+  }
+
+
+  map[goal_row][goal_col].cell_status = 2;
+
+  map[start_row][start_col].cell_status = 0;
+  map[start_row][start_col].move_cost  = 0;
+  map[start_row][start_col].heur_cost = heuristics(start_row, start_col, goal_row,  goal_col);
+  map[start_row][start_col].parent_node_row = start_row;
+  map[start_row][start_col].parent_node_col = start_col;
+  map[start_row][start_col].cost = heuristics(start_row, start_col, goal_row,  goal_col) + 0 ;
+  map[start_row][start_col].visited = true;
+  
+  int p;
+  int q;
+  
+  int current_node_row = start_row;
+  int current_node_col = start_col;
+  float cost = 0 ;
+  
+
+  
+  while(!((current_node_row == goal_row) && (current_node_col == goal_col)))
```

Comment:
move this check to a seperate function called check_end_condition or something similar